### PR TITLE
[Ready]Bluespace research updates 

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
balance: Advanced Bluespace Research has been split in Bluespace Travel and Miniaturized Bluespace Research, with the former relating to static teleporters and the latter to small devices and stock parts.
balance: Hand teleporters are now protolathe-buildable in research and engineering, requiring Miniaturized Bluespace Research.
add: Added Quantum Keycards, devices that can link to a quantum pad, and can be used on any other quantum pad to teleport to its linked pad.
tweak: Bluespace Power Technology now requires Applied Bluespace Research instead of Advanced Bluespace Research.
/:cl:

Made bluespace research a little bit more modular instead of a 15000 point dam that unlocks everything at once.

Quantum keycards are the quantum version of hand teleporters, making qpad travel more free for its users, or allowing key-restricted qpads for secure locations.

Hand teleporters being printable seems reasonable to me, as there are more powerful devices available already (like portal guns, or bags of holding). It is a traitor objective, but then again reflector vests can be ordered by cargo, so it's not the first objective that can be replaced ingame.

### CODE REVIEW

**Comment by davethwave** [comment]
Seems good.

**Comment by Dorsisdwarf** [comment]
i like the idea of setting up a qpad that just fucking dumps you into a black hole or a kill chamber then having a qcard linked to your fortress so if you're being chased you use the card and the guy chasing you ends up getting fucked

**Comment by Tlaltecuhtli** [comment]
remove hand tele from antag obj

File: code/modules/research/designs/misc_designs.dm [hunk 1]
```diff
@@ -286,6 +286,26 @@
...
+	build_path = /obj/item/quantum_keycard
+	category = list("Equipment")
+	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE | DEPARTMENTAL_FLAG_ENGINEERING
+	
+/datum/design/hand_tele
+	name = "Hand Teleporter"
+	desc = "Allows for the construction of a hand teleporter."
+	id = "hand_tele"
+	build_type = PROTOLATHE
+	materials = list(MAT_GLASS = 500, MAT_METAL = 500, MAT_GOLD = 500, MAT_BLUESPACE = 2000, MAT_DIAMOND = 2000)
```
Comment (octareenroon91) [inline code/modules/research/designs/misc_designs.dm]: Diamond to 3000 maybe.

Comment (davethwave) [reply code/modules/research/designs/misc_designs.dm]: Nah, would make sense to bump up the bluespace crystal cost instead of the diamond one.

Comment (octareenroon91) [reply code/modules/research/designs/misc_designs.dm]: Both 2100 would be fine: The threshold is 2000 units per sheet and I'm suggesting that the hand tele should cost just a bit more on paper than 1 sheet.

**Comment by BlueNothing** [comment]
Or don't. This sounds like it'll be cool either way, though. So are portal guns in Travel or Miniaturized?

**Comment by actioninja** [comment]
I like this, the 15000 point gate for so much was always annoying, doubly so when some fuck keeps wasting the points.

**Comment by subject217** [comment]
Being able to manufacture hand teles is really powerful, likely too powerful, but it'll be fun for a month or so. The techwebs changes are good.

**Comment by Garen7** [comment]
boy i cant wait to teleport into armory to steal guns everyshift with my own personal hand tele

**Comment by XDTM** [comment]
It's the same research as a portal gun, which i think is more versatile, and if you want to visit beacons you still need a static teleporter to target them.

**Comment by subject217** [comment]
@Garen7 hack open teleporter room at roundstart bam hand tele without techwebs time gate 

the op aspect is that everyone and their brother will have one

**Comment by BlueNothing** [comment]
Is this still in the pipes?

**Comment by ExcessiveUseOfCobblestone** [comment]
Is the hand tele really that strong compared to the portal gun?

**Comment by AnturK** [comment]
Code looks fine. Handtele is one of the few unique items with inherent value left. And was intentionally kept as such for a while now. Remove the addition from this PR please. We can argue about replacing it with something another time.

**Comment by XDTM** [comment]
Alright

**Comment by BlueNothing** [comment]
So... remove the hand-tele and it should be good?

**Comment by XDTM** [comment]
oh yeah i forgot to actually do it, should be fine now

**Comment by AnturK** [comment]
Oof didn't notice changelog was left as is.

**Comment by XDTM** [comment]
oh damn my bad

**Comment by SpaceManiac** [comment]
## WARNING: Invalid research prerequisite node with ID adv_bluespace detected in node One-point Bluespace-gravitational Manipulator[gravity_gun] removed. in code/modules/research/techweb/__techweb_helpers.dm at line 88 src:  usr: .
## WARNING: Invalid research prerequisite node with ID adv_bluespace detected in node EXOSUIT: Phazon[mecha_phazon] removed. in code/modules/research/techweb/__techweb_helpers.dm at line 88 src:  usr: .
## WARNING: Invalid research prerequisite node with ID adv_bluespace detected in node Exosuit Module (Teleporter Module)[mech_teleporter] removed. in code/modules/research/techweb/__techweb_helpers.dm at line 88 src:  usr: .
## WARNING: Invalid research prerequisite node with ID adv_bluespace detected in node Exosuit Module (Localized Wormhole Generator)[mech_wormhole_gen] removed. in code/modules/research/techweb/__techweb_helpers.dm at line 88 src:  usr: .

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -171,18 +171,26 @@
...
-	prereq_ids = list("practical_bluespace", "high_efficiency")
-	design_ids = list("bluespace_matter_bin", "femto_mani", "triphasic_scanning", "tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console",
-	"teleconsole", "bag_holding", "bluespace_crystal", "wormholeprojector", "bluespace_pod")
-	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 15000)
+/datum/techweb_node/bluespace_travel
+	id = "bluespace_travel"
+	display_name = "Bluespace Travel"
+	description = "Application of Bluespace for static teleportation technology."
+	prereq_ids = list("practical_bluespace")
+	design_ids = list("tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console", "bluespace_pod")
```
Comment (Dennok) [inline code/modules/research/techweb/all_nodes.dm]: Here is lost "teleconsole"

**Comment by octareenroon91** [comment]
(  #40487 )

**Pull request merged**